src/emit/solana/target.rs

```diff
@@ -1707,11 +1708,7 @@ impl<'a> TargetRuntime<'a> for SolanaTarget {
                     "key_pointer",
                 );
 
-                binary.builder.build_load(
-                    binary.address_type(ns),
-                    key_pointer.into_pointer_value(),
-                    "key",
-                )
+                key_pointer.as_basic_value_enum()
```

Comment:
This is concerning. The scratch buffer is a global buffer which gets overwritten by lots of syscalls/code. We should have a comment explaining that this needs to be taken into account for codegen.

Reply:
I'll add the comment. This is what already happens with `Expression::ReturnData`, isn't it?

Reply:
@seanyoung The comment should go in the substrate file, right? You commented in the solana target.

Reply:
yes, you're right. Also please comment the returndata one if it isn't already

Reply:
Fixed!